packages/@aws-cdk/aws-codebuild/lib/artifacts.ts

```diff
@@ -142,9 +149,10 @@ class S3Artifacts extends Artifacts {
         location: this.props.bucket.bucketName,
         path: this.props.path,
         namespaceType: this.props.includeBuildId === false ? 'NONE' : 'BUILD_ID',
-        name: this.props.name,
+        name: this.props.name == null ? project.node.uniqueId : this.props.name,
```

Comment:
Did you test that this is necessary - we can't just leave this as `undefined`? `name` is optional in `CfnProject.ArtifactsProperty`: https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-codebuild.CfnProject.ArtifactsProperty.html#name

Reply:
I didn't - I'll do so today and set it to undefined if that works.

Reply:
Done, hopefully the integration tests pass!

Reply:
Just to make sure we're on the same page - by default, the integration tests act like snapshot tests. You need to specifically run the `npm integ` target to have them deployed. More info here: https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md#integration-tests

Reply:
Ah, okay! I'll run those this evening and let you know the outcome...

Reply:
Just ran the tests against my own account and all the `aws-codebuild` integration tests provision and tear down correctly, no errors visible in the CloudFormation console or build logs!